### PR TITLE
Refactors talking indicators + new talking ind's for *some* species

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Refactors the talking indicator to not always use default0, replaces the slime talking sprite (Golgor#7400 on discord), and adds bee, ethereal, and golem sprites.

## Why It's Good For The Game
variety, makes a central mechanic actually function

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

default
image

bee
image

ooze
image

slime
image

golem
image

</details>

## Changelog
:cl:
refactor: talking sprite indicators dont always use default0 now
spriteadd: more talking sprite indicators!
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
I would prefer these to be from tongue type rather than species.

**Comment by etherware-novice** [comment]
coloured indicators?

**Comment by EvilDragonfiend** [comment]
Good. Now you can visually see how they speak through which tongue verb type.



> coloured indicators?

Since the idea is based on how they speak, emote indicate should be exceptional from this as emote doesn't have anything to identify (unless you add emote only indicator)

**Comment by EvilDragonfiend** [comment]
Although it is bad that you can identify their species even if they are obscured in space... but I don't have any better idea for meta-ish playing of that.

File: code/modules/mob/typing_indicator.dm [hunk 1]
```diff
@@ -68,11 +66,14 @@ Both the say/me wrappers and cancel_typing remove the typing indicator.
...
 ///Human Typing Indicators///
 /mob/living/create_typing_indicator()
-	if(!typing_indicator && stat == CONSCIOUS) //Prevents sticky overlays and typing while in any state besides conscious
-		add_overlay(GLOB.typing_indicator)
-		typing_indicator = TRUE
+	if(typing_indicator || stat != CONSCIOUS) //Prevents sticky overlays and typing while in any state besides conscious
+		return
+	var/obj/item/organ/tongue/voice = getorganslot(ORGAN_SLOT_TONGUE)
+	var/sprite_used = voice?.bubble_sprite || bubble_icon || "default"  // if theres no bubble default to default
+	typing_indicator = mutable_appearance('icons/mob/talk.dmi', "[sprite_used]0", -TYPING_LAYER)
```
Comment (itsmeow) [inline code/modules/mob/typing_indicator.dm:80]: just remove the 0 from the icon states

File: code/modules/mob/typing_indicator.dm [hunk 2]
```diff
@@ -68,11 +66,14 @@ Both the say/me wrappers and cancel_typing remove the typing indicator.
 
 ///Human Typing Indicators///
 /mob/living/create_typing_indicator()
-	if(!typing_indicator && stat == CONSCIOUS) //Prevents sticky overlays and typing while in any state besides conscious
-		add_overlay(GLOB.typing_indicator)
-		typing_indicator = TRUE
+	if(typing_indicator || stat != CONSCIOUS) //Prevents sticky overlays and typing while in any state besides conscious
+		return
+	var/obj/item/organ/tongue/voice = getorganslot(ORGAN_SLOT_TONGUE)
```
Comment (itsmeow) [inline code/modules/mob/typing_indicator.dm:71]: it should be stored on the species datum, not the tongue organ.

File: code/modules/mob/mob_defines.dm [hunk 3]
```diff
@@ -220,5 +220,5 @@
 	///Override for sound_environments. If this is set the user will always hear a specific type of reverb (Instead of the area defined reverb)
 	var/sound_environment_override = SOUND_ENVIRONMENT_NONE
 
-	///Currently possesses a typing indicator icon
-	var/typing_indicator = FALSE
+	/// The current typing indicator used (usually false)
+	var/mutable_appearance/typing_indicator = FALSE
```
Comment (itsmeow) [inline code/modules/mob/mob_defines.dm:224]: make this a global list that stores a mapping of `sprite_used` -> `mutable_appearances` so every mob doesn't have to initialize their own mutable_appearance.

Then just use `GLOB.typing_indicators[sprite_used]` etc

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/mob/typing_indicator.dm (hunk 1 above)
Comment (itsmeow) [reply code/modules/mob/typing_indicator.dm:80]: i see now that there are states with ? and !, but they're unused. odd.

File: code/modules/mob/typing_indicator.dm (hunk 2 above)
Comment (itsmeow) [reply code/modules/mob/typing_indicator.dm:71]: well, maybe. it's up for debate. see what others think.

Comment (itsmeow) [reply code/modules/mob/typing_indicator.dm:71]: Oh. I just noticed the previous changes. Okay. Whatever, probably fine as is.

**Comment by itsmeow** [comment]
In the situation where someone without an ID equipped would say "Unknown", the indicator should revert to default.

**Comment by EvilDragonfiend** [comment]
> In the situation where someone without an ID equipped would say "Unknown", the indicator should revert to default.

It is not consistent since voice is an exposed information

> Also, if you're using a voice changer, it should lookup in the manifest and try to find the matching species perhaps or otherwise revert to default

Unlesa voice changer supports tongue verb disguise, this is not consistent too, I believe. The problem is only when you talk in spcae where you can't hear their verb. I would prefer to see the indicator becomes default when there's no pressure around the sayer.

**Comment by etherware-novice** [comment]
alright uh im letting yall know now im going on vacation for a few days so i wont be able to workk on this for a while

**Comment by EvilDragonfiend** [comment]
> itsmeow 1 hour ago
it should be stored on the species datum, not the tongue organ.
> itsmeow 1 hour ago
well, maybe. it's up for debate. see what others think.
> itsmeow 1 hour ago
Oh. I just noticed the previous changes. Okay. Whatever, probably fine as is.

So about this, this should be interpreted in two ways.
1. How do they look when they speak
2. How do they sound when they speak

The PR suggested 1 at first, but I didn't like feel it is very necessary since we already can recognise how they look visually. Recognising their mouth shape wouldn't be needed.
While my suggestion was 2, so that you can visually notice if they have different tongue. That is very intuitive when you don't look at chat logs well.
But this is bad as you can identify them visually even if you didn't see their appearance, so hiding their indicator to default would be necessary when they are unknown.

I think we might need an indicator for emote and unknown only, so make it more consistent.

**Pull request closed without merging**